assets/js/googlesitekit/widgets/components/WidgetAreaRenderer.js

```diff
@@ -106,19 +106,18 @@ const WidgetAreaRenderer = ( { slug } ) => {
 	const widgetArea = useSelect( ( select ) => select( STORE_NAME ).getWidgetArea( slug ) );
 	const widgets = useSelect( ( select ) => select( STORE_NAME ).getWidgets( slug ) );
 
-	// State handled by WidgetRenderer instances, based on whether the widget
-	// renders content or `null`.
-	const [ activeWidgets, setActiveWidgets ] = useState( {} );
+	// Verify that widgets are active (do not render null)
+	const isActiveWidget = ( widget ) =>
+		widgets.find( ( item ) => item.slug === widget.slug ) &&
+		typeof widget.component === 'function' &&
+		widget.component( {} );
```

Comment:
@eugene-manuilov This is a good concern to call out as this will fail if presented with a class component. Using `createElement` directly is unnecessary, we could avoid the import by using JSX directly (this gets changed to createElement under the hood, it's syntatical sugar) via `<widget.component />`. Unfortunately the return value of `createElement` is an object which does not contain the return value of the component, which is what we need to assert against.

Reply:
You are right, @caseymorrisus. What about using [`renderToStaticMarkup()`](https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup)? It will render a widget and return a string that we can check in the condition.